server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java

```diff
@@ -0,0 +1,48 @@
+package org.greenplum.pxf.plugins.json;
+
+import org.greenplum.pxf.api.model.ProtocolHandler;
+import org.greenplum.pxf.api.model.RequestContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+
+/**
+ * Implementation of ProtocolHandler for json protocol.
+ */
+public class JsonProtocolHandler implements ProtocolHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JsonProtocolHandler.class);
+    private static final String HCFS_FILE_FRAGMENTER = "org.greenplum.pxf.plugins.hdfs.HdfsFileFragmenter";
+
+    @Override
+    public String getFragmenterClassName(RequestContext context) {
+        String fragmenter = context.getFragmenter(); // default to fragmenter defined by the profile
+        if (useMultilineJson(context)) {
+            fragmenter = HCFS_FILE_FRAGMENTER;
+        }
+        LOG.debug("Determined to use {} fragmenter", fragmenter);
+        return fragmenter;
+    }
+
+    @Override
+    public String getAccessorClassName(RequestContext context) {
+        return context.getAccessor();
+    }
+
+    @Override
+    public String getResolverClassName(RequestContext context) {
+        return context.getResolver();
+    }
+
+
+    private boolean useMultilineJson(RequestContext context) {
+        String identifier = context.getOption("identifier");
+
+        if (isNotEmpty(identifier)) {
+            return true;
+        }
+
+        return false;
```

Comment:
```suggestion
        return isNotEmpty(context.getOption("identifier"));
```
we can make this a 1-liner or even embed into the calling function